packages/syft/src/syft/service/worker/worker_image.py

```diff
@@ -18,64 +13,67 @@
 from ...types.syft_object import SYFT_OBJECT_VERSION_1
 from ...types.syft_object import SyftObject
 from ...types.uid import UID
-from ..response import SyftError
-from ..response import SyftSuccess
-
-
-def parse_output(log_iterator: Iterator) -> str:
-    log = ""
-    for line in log_iterator:
-        for item in line.values():
-            if isinstance(item, str):
-                log += item
-            elif isinstance(item, dict):
-                log += json.dumps(item)
-            else:
-                log += str(item)
-    return log
-
-
-@serializable()
-class ContainerImageRegistry(SyftBaseModel):
-    url: str
-    tls_enabled: bool
-
-    def from_url(cls, full_str: str):
-        return cls(url=full_str, tls_enabled=full_str.startswith("https"))
-
-    def __hash__(self) -> int:
-        return hash(self.url + str(self.tls_enabled))
-
-    def __str__(self) -> str:
-        return self.url
+from .image_registry import SyftImageRegistry
 
 
 @serializable()
 class SyftWorkerImageTag(SyftBaseModel):
-    registry: Optional[ContainerImageRegistry]
     repo: str
     tag: str
+    registry: Optional[SyftImageRegistry | str]
 
     @classmethod
-    def from_str(cls, full_str: str) -> Self:
-        repo_url, tag = full_str.rsplit(":", 1)
-        args = repo_url.rsplit("/", 2)
+    def from_registry(cls, tag: str, registry: SyftImageRegistry) -> Self:
+        """Build a SyftWorkerImageTag from Docker tag & a previously created SyftImageRegistry object."""
+        registry_str, repo, tag = SyftWorkerImageTag.parse_str(tag)
+
+        # if we parsed a registry string, make sure it matches the registry object
+        if registry_str and registry_str != registry.url:
+            raise ValueError(f"Registry URL mismatch: {registry_str} != {registry.url}")
+
+        return cls(repo=repo, tag=tag, registry=registry)
+
+    @classmethod
+    def from_str(cls, tag: str) -> Self:
+        """Build a SyftWorkerImageTag from a pure-string standard Docker tag."""
+        registry, repo, tag = SyftWorkerImageTag.parse_str(tag)
+        return cls(repo=repo, registry=registry, tag=tag)
+
+    @staticmethod
+    def parse_str(tag_str: str):
```

Comment:
nitpick: `tag: str` instead of `tag_str: str`. cleaner and more consistent with `from_str`. I would even prefer `def parse(tag: str)` but it's your call.

Reply:
Also this is a helper function we used internally should it be `_parse_str`/`_parse` instead?

Reply:
> Also this is a helper function we used internally should it be `_parse_str`/`_parse` instead?

It's fine to have this as a pub fn... I found myself using this at one point in worker image service.

Reply:
It's perfectly fine to use underscored methods elsewhere in the code base. It doesn't mean you can't use it outside of its class. It just means it's not an end user public API.